src/reducer/playlistReducer.js

```diff
@@ -0,0 +1,36 @@
+
+const playlistReducer=(state,{type,payload})=>{
+    switch(type){
+        case "FETCHING":
+            return{
+                ...state,
+                loading:true
+            };
+        case "FETCHED":
+            return{
+                ...state,
+                loading:false,
+                playlists:payload
```

Comment:
From your code what I understood that, you are using loader when a network call is being made, I guess for that using `reducer` would be a lot, you can just use `useState` for it and send a dispatch for playlist only when you get all of your playlists as response and then add them to your context.